s3/replication/build_and_test.sh

```diff
@@ -0,0 +1,195 @@
+# s3replicator and s3replicationmanager.
+#
+# Usage : sh ./build_and_test.sh
+# 
+
+BUILD_SCRIPT_PATH=$(readlink -f "$0")
+# Replication directory
+REPLICATION_DIR=$(dirname "$BUILD_SCRIPT_PATH")
+# Project root source directory
+MULTISITE_ROOT_DIR="$REPLICATION_DIR/../.."
+# Virtual environment directory
+REPLICATION_VENV=$REPLICATION_DIR/s3env/bin
+
+echo "Project root source directory : "$MULTISITE_ROOT_DIR
+echo "Replication directory :" $REPLICATION_DIR
+echo "Virtual environment directory : "$REPLICATION_VENV
+
+# Module directory
+S3_REPLICATION_COMMON_SRC_DIR=$REPLICATION_DIR/common/src
+S3_REPLICATION_MANAGER_SRC_DIR=$REPLICATION_DIR/manager/src
+S3_REPLICATOR_SRC_DIR=$REPLICATION_DIR/replicator/src
+
+# Test directory
+S3_REPLICATION_COMMON_TEST_DIR=$REPLICATION_DIR/common/tests/system
+S3_REPLICATION_MANAGER_TEST_DIR=$REPLICATION_DIR/manager/tests/system
+S3_REPLICATOR_TEST_DIR=$REPLICATION_DIR/replicator/tests/system
+
+echo "Module directories -"
+echo "Common module directory :" $S3_REPLICATION_COMMON_SRC_DIR
+echo "Manager module directory :" $S3_REPLICATION_MANAGER_SRC_DIR
+echo "Replicator module directory :" $S3_REPLICATOR_SRC_DIR
+
+# Helper method to stop background services.
+stop_replication_services()
+{
+    systemctl stop s3_in_memory.service
+    systemctl stop s3replicationmanager.service
+    systemctl stop s3replicator.service
+}
+
+# Active an environment
+python3 -m venv $REPLICATION_DIR/s3env
+source $REPLICATION_DIR/s3env/bin/activate
+
+# Check and download dependencies
+pip3 install --upgrade setuptools
+
+cd $REPLICATION_DIR
+make install
+
+# Create and copy files
+S3_IN_MEMRORY_SVC_FILE=$S3_REPLICATION_COMMON_SRC_DIR/systemd/s3_in_memory.service
+MANGER_SVC_FILE=$S3_REPLICATION_MANAGER_SRC_DIR/systemd/s3replicationmanager.service
+REPLICATOR_SVC_FILE=$S3_REPLICATOR_SRC_DIR/systemd/s3replicator.service
+
+# Systemd service files
+SYSD_S3_IN_MEMORY_SVC_FILE=/etc/systemd/system/s3_in_memory.service
+SYSD_REPLICATION_MANAGER_SVC_FILE=/etc/systemd/system/s3replicationmanager.service 
+SYSD_REPLICATOR_SVC_FILE=/etc/systemd/system/s3replicator.service
+
+# Create creadential file in home directory with below path.
+# ~/.cortxs3/credentials.yaml
+CORTX_S3_CRED_FILE=~/.cortxs3/credentials.yaml
+CORTX_S3_DIR="$(dirname "$CORTX_S3_CRED_FILE")"
+if [ -f "$CORTX_S3_CRED_FILE" ]; then
+    echo "$CORTX_S3_CRED_FILE exist!"
+else
+    if [ -d $CORTX_S3_DIR]; then
+        echo "$CORTX_S3_DIR exist!"
+    else
+        mkdir $CORTX_S3_DIR
+    fi
+    touch $CORTX_S3_CRED_FILE
+    # Placeholder for keys to run against in-memory s3 server.
+    echo "access_key: '<your access_key>'" > $CORTX_S3_CRED_FILE
+    echo "secret_key: '<your secret_key>'" >> $CORTX_S3_CRED_FILE
+fi
+
+rm -f $SYSD_S3_IN_MEMORY_SVC_FILE 
+rm -f $SYSD_REPLICATION_MANAGER_SVC_FILE
+rm -f $SYSD_REPLICATOR_SVC_FILE
+
+# Strip hardcoded lines
+sed "/ExecStart/d" $S3_IN_MEMRORY_SVC_FILE > $SYSD_S3_IN_MEMORY_SVC_FILE
```

Comment:
use something like this. 

sed -i '/REPLICATION_VENV_PLACEHOLDER/$REPLICATION_VENV/'

Reply:
done.